### PR TITLE
Hamming: Updates canonical-data.json to comply with schema

### LINKED ISSUE #812: hamming: Update expected errors in canonical data
From: https://github.com/exercism/x-common#test-data-format-canonical-datajson
> if an error is expected (because the input is invalid, or any other reason), the value at "expected" should be an object containing exactly one property, "error", whose value is a string.
>     The string should explain why the error would occur.

But the [hamming canonical data](https://github.com/exercism/x-common/blob/master/exercises/hamming/canonical-data.json#L113) has an expected value of  `-1`

Todo: 

Update https://github.com/exercism/x-common/blob/master/exercises/hamming/canonical-data.json to use error objects for the expected value in the error cases:  
e.g.  `"expected"   : { "error": "some helpful message" }`

### PR DESCRIPTION
**Pull request by m-a-ge**
Should resolve #812

### CODE REVIEW

**Comment by m-a-ge** [comment]
@ErikSchierboom, thanks for pointing out to the issue. 
My bad (still new to the project).
I'll update the PR later toady

**Comment by ErikSchierboom** [comment]
@m-a-ge No problem! Thanks for taking the effort to do a PR.

File: exercises/hamming/canonical-data.json [hunk 1]
```diff
@@ -110,14 +110,14 @@
       "property": "distance",
       "strand1": "AATG",
       "strand2": "AAA",
-      "expected": -1
+      "expected": {"error": "leftStrand and rightStrand must be of equal length"}
```
Comment (stevejb71) [inline exercises/hamming/canonical-data.json]: ```leftStrand``` looks a bit like a Java parameter name. How about 'left strand'? Or 'strands are of different lengths'?

Comment (m-a-ge) [inline exercises/hamming/canonical-data.json]: It makes sense, will change it as well

**Comment by m-a-ge** [comment]
@ErikSchierboom, I've made requested changes, please verify 👮

**Pull request merged**